src/components/ListItem.jsx

```diff
@@ -1,5 +1,66 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase.js';
+import { useState, useEffect, useCallback } from 'react';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils/dates.js';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listPath, item }) {
+	const purchasedOneDayAgo = useCallback(() => {
+		if (item.dateLastPurchased !== null) {
+			const timeDiff = Date.now() - item.dateLastPurchased.seconds * 1000;
+			if (timeDiff <= ONE_DAY_IN_MILLISECONDS) {
+				return true;
+			} else {
+				return false;
+			}
+		} else {
+			return false;
+		}
+	}, [item.dateLastPurchased]);
+
+	const [isPurchased, setIsPurchased] = useState(false);
+	const [isChecked, setIsChecked] = useState(purchasedOneDayAgo);
+	function changeHandler(e) {
+		setIsPurchased(!isPurchased);
+		setIsChecked(!isChecked);
+	}
+
+	useEffect(() => {
+		async function purchaseItem() {
+			if (isPurchased) {
+				try {
+					await updateItem(listPath, item.id);
+				} catch (error) {
+					alert(error.message);
+				}
+			}
+		}
+		purchaseItem();
+	}, [isPurchased, item.id, listPath]);
```

Comment:
I think it would be a little bit be if you removed the `useEffect` hook and just added the logic into the `changeHandler` function, so that you are not introducing additional re-rendering logic into the app. You can probably also get rid of the `isPurchased` state as well, since we know if the item has been purchased if there is a `lastPurchaseDate`

Reply:
I tried to refactor it with a callback function as the second argument of setState, but couldn't quite wrap my head around it within the time constraints of this sprint. The way I settled on looks a little janky, but works for now 😅